Modules/Core/Common/include/itkProcessObject.h

```diff
@@ -880,6 +908,18 @@ class ITKCommon_EXPORT ProcessObject : public Object
    * It prevents infinite recursion when pipelines have loops. */
   bool m_Updating;
 
+  /** Internal variables used in progress reporting. */
+  bool m_MultiThreaderUpdatesProgress =
+    true; // coarse-grained progress reporting does not require cooperation of filters
+  std::atomic<SizeValueType>            m_CurrentPixelCount{ 0 };
+  IdentifierType                        m_TotalPixelCount{ 0 };
+  IdentifierType                        m_PreviousProgressReport{ 0 }; // pixel count during last report
+  std::chrono::steady_clock::time_point m_LastProgressReport =
+    std::chrono::steady_clock::now(); // time during last report
+  std::chrono::milliseconds m_DesiredUpdateInterval =
+    std::chrono::milliseconds(100);                               // TODO: have a statically settable global default;
+  std::thread::id m_CallingThreadID = std::this_thread::get_id(); // we can only report from this thread
```

Comment:
This would be the thread of construction. I believe the progress is only for the `GenerateData` method. The best place to get the thread id of the pipeline execution might be right before the GenerateData method is called in `itk::ProcessObject::UpdateOutputData`. Then the variable could be made private and some accessor like "IsPipelineThread" could be used.

Reply:
Good catch.